app/libs/triggers/pull_request.rb

```diff
@@ -35,6 +35,8 @@ def create_and_merge!
       @pull_request = @new_pull_request.update_or_insert!(create.value!)
     end
 
+    return GitHub::Result.new { @pull_request } if @pull_request.closed?
```

Comment:
Shouldn't this be further up? `@pull_request` is fetched before, and it seems like we're early returning if it's already marked as closed in the db, isn't that right?

Reply:
This is the common logic between existing PR and new PR. If the response we fetch or that is returned to us after creation says the PR is closed, we don't need to call merge. Even an existing PR should be merged if not already closed in VCS.